app/lib/memory_repository.rb

```diff
@@ -192,10 +192,13 @@ def get_all_revisions
 
   # Static method: Yields an existing Memory repository and closes it afterwards
   def self.access(connect_string)
-    repository = MemoryRepository.open(connect_string)
-    yield repository
-  ensure
-    repository.close
+    raise 'Respository does not exist' unless MemoryRepository.repository_exists? connect_string
+    begin
```

Comment:
You can remove the begin and do `repository&.close` in the ensure.

Reply:
same comment as above